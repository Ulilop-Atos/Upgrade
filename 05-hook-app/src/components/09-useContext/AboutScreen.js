import { useContext } from "react";
import { UserContext } from "./UserContext";

export const AboutScreen = () => {
  const { user, setUser } = useContext(UserContext);

  const handleClick = () => {
    setUser({});
  };

  return (
    <>
      <h1>AboutScreen</h1>
      <hr />
      <pre className="container">
        {JSON.stringify(user,null,2)}
      </pre>

      <button
        className="btn btn-primary"
        onClick={handleClick}
      >
        Logout
      </button>
    </>
  )
}

export default AboutScreen;