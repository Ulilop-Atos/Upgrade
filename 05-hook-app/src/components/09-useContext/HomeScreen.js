import { useContext } from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {

  const { user } = useContext(UserContext);

  return (
    <>
      <h1>HomeScreen</h1>
      <hr />
      <pre className="container">
        {JSON.stringify(user,null,2)}
      </pre>
    </>
  )
}

export default HomeScreen;