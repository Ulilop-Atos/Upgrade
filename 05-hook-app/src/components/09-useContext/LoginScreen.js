import { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {

  const {setUser} = useContext(UserContext);

  const userExample = {
    id: 123,
    name: 'Ferdinand',
    email: 'ferdinand@mail.com',
  };

  return (
    <>
      <h1>LoginScreen</h1>
      <hr />
      <button
        className="btn btn-primary"
        onClick={() => setUser(userExample)}
      >
        Login
      </button>
    </>
  )
}

export default LoginScreen;