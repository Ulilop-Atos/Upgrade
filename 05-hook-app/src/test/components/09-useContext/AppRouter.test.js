import { mount } from "enzyme";
import AppRouter from "../../../components/09-useContext/AppRouter";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('Test on <AppRouter />', () => {

  const user = {
    id: 0, 
    name: 'Test Name',
    email: 'test@mail.com',
  }

  const setUser = jest.fn();

  const wrapper = mount(
    <UserContext.Provider value={{user, setUser}}>
      <AppRouter />
    </UserContext.Provider>
  );
   
  test('should display properly', () => {
    expect(wrapper).toMatchSnapshot(); 
  });
})