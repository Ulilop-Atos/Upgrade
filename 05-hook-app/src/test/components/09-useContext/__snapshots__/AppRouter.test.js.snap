// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <AppRouter /> should display properly 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Hooks Context App
      </a>
      <button
        aria-controls="navbarNavAltMarkup"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarNavAltMarkup"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            active="active"
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            active="active"
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            active="active"
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    Array [
      <h1>
        HomeScreen
      </h1>,
      <hr />,
      <pre
        className="container"
      >
        {
  "id": 0,
  "name": "Test Name",
  "email": "test@mail.com"
}
      </pre>,
    ]
  </div>,
]
`;
